### PR TITLE
new recommendations page

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
New recommendations page for the new diagnostic reports.

## WHY
Needed a style update to match the existing ones, and we also needed to update the backend logic to account for the fact that new pre/post tests do not necessarily use a specific unit id.

## HOW
Update the backend to find the relevant sessions with no unit, and also to batch assign Quill Lessons recommendations; whole new frontend.

### Screenshots
Screen Shot 2021-11-04 at 10 31 36 AM
Screen Shot 2021-11-04 at 10 31 46 AM
Screen Shot 2021-11-04 at 11 44 55 AM
Screen Shot 2021-11-04 at 11 45 04 AM

### Notion Card Links
https://www.notion.so/quill/ea241f31a84c4b79ad56e83e99f7ee93?v=0c466a1f8c3846db94b8c959984cbb9d#2ffa6ee8bb904bdea6821302ef2c685b

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | YES
Have you deployed to Staging? | NO - waiting for Tom to be able to review all at once
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

**Review by anathomical — APPROVED** [review]
Looks good!

File: services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/recommendationsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  const [stickyTableStyle, setStickyTableStyle] = React.useState({
+    position: "fixed",
+    top: 0,
+    left: 0,
+    right: 0,
+    zIndex: 3
+  })
+
+  const handleScroll = React.useCallback(({ top, bottom, left, right, }) => {
+    if (top <= 0 && bottom > 92) {
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/recommendationsTable.tsx:116]: I think a comment explaining why we're using these values would be nice, like referencing the name of the class for a container just so we can know to make changes in both places should the need arise

**Review by cissyxyu — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/index.jsx [hunk 2]
```diff
@@ -36,13 +36,14 @@ class DiagnosticReports extends React.Component {
...
 		}
-    if (this.onPageThatHandlesItsOwnRendering()) {
+    if (!this.onPageThatHandlesItsOwnRendering()) {
 			this.getStudentAndActivityData(nextParams);
 		}
 	};
 
   onPageThatHandlesItsOwnRendering = () => {
     const { location, } = this.props
+    console.log('do you think I am here', ['/activity_packs', '/not_completed', '/diagnostics'].indexOf(location.pathname) !== -1 || location.pathname.includes('/diagnostics'))
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/index.jsx]: leftover console log (we have very similar naming habits haha)

File: services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/index.jsx [hunk 3]
```diff
@@ -75,6 +76,7 @@ class DiagnosticReports extends React.Component {
 		let ajax = this.ajax;
     const that = this
     const p = params || this.parseParams(this.props.location.pathname);
+    console.log('is this getting called')
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/index.jsx]: and here too

File: services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/recommendationsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+      />)
+    })
+  }
+  return <tr key={name}>{firstCell}{selectionCells}</tr>
+}
+
+const RecommendationsTable = ({ recommendations, students, selections, previouslyAssignedRecommendations, setSelections, }: RecommendationsTableProps) => {
+  const [isSticky, setIsSticky] = React.useState(false);
+  const tableRef = React.useRef(null);
+  const [stickyTableStyle, setStickyTableStyle] = React.useState({
```
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/recommendationsTable.tsx]: would be nice to have some type definitions here

**Review by eadams17 — APPROVED** [review]
looks good! just a few small comments

**Pull request merged**